 uint8 count
 float32 x
 float32 y
 float32 z
 float32 alpha
 float32 beta
float32 gamma
float32 newX
  float32 newY
  float32 newZ                                        
 float32 distance
